pam/model.go

```diff
@@ -13,17 +13,17 @@ import (
 var debug string
 
 // state represents the stage object.
-type stage int
+type stage = authd.Stage
```

Comment:
Well, the reason for this is sharing the very same value between the pam module, and the GDM json protocol.

I initially was doing this by serializing the stage value to JSON, but it was just leading to repetitive code (as shortly explained in the commit message of https://github.com/ubuntu/authd/pull/155/commits/7007d0dfca3d2cf9ae6dee64850effc554411257).

So, we can avoid having this in the proto file, but it would imply still exposing a type and then converting it, while having a proto-type make things cleaner at the end of the day.

Reply:
> One way would be to have a separate proto files inside the pam/ directory directly.

We do have that already, but it also [references the "parent" `authd.proto`](https://github.com/ubuntu/authd/blob/main/pam/gdm/gdm.proto#L6), since they share the same data. And while an enum is just a simple case for this, structs such as the UI layouts are something that is better to share.

This can change if authd.proto needs to change (while the json one does not), but for the current time this avoids lots of duplication.

> you don’t generate json from it, do you?. Is it a string version of it?

Yeah, we do generate json [using the proto](https://github.com/ubuntu/authd/blob/main/pam/gdm/protocol.go). I initially was all doing things manually, but then I thought this was more future-proof, and avoided potential mistakes.

> If so, we can have that using the Stringer generator.

Can be a string too, but I wouldn't involve Stringer just for these few cases, I did in my previous iteration of this, but then I switched using protobuf for JSON too, and it simplified things.

Reply:
If you generate that using the proto (that’s the part that was merged during my holidays), then, that makes sense. Just ensure to comment it properly in the proto file that this is only for the PAM module to the external UI communication.

I wonder, if we should remove it from the parent authd.proto, and have another proto in `pam/` just for it. Then, the gdm one can include this one. At least, it doesn’t end up in the parent `authd.proto` that way.

Reply:
However this commit initially was just as https://github.com/3v1n0/authd/commit/4c70f2da1fd25ff00de0ced0aed80baabf7cdcd4 that was enough to cleanup the code, but then it looked smarter to just define a type that is shared everywhere and that recognized by the proto both in the go and JS sides of it.

It could probably just be an int (even though I like json strings at dev time, since they're easier to debug), but I liked the idea to have a single place where to maintain the type.

Reply:
Ok, new proto file added... It's probably tiny now, but I feel I'll add few more things in the future so I think it's a better way.

Reply:
Oh, actually... This required to use another package... So, since I had in mind something like this for a long time I decided to take the occasion to move the model code into a non-main package so that we only have the real main-files into that.

I started with https://github.com/ubuntu/authd/pull/155/commits/503f0221db63bd5a135d481fcc8a68b3eea6e0cb (the new package name really needs love :)), but just to give an idea...

Ideally I'd like to change things in a way that we don't mix the models data with the main one that is only required to generate the test binaries and the shared library.

Reply:
I guess this conversation can be close too, right?

I'd do it feel free to revert if that's not the case...

Reply:
with the separate model, agreed.